CodePush.m

```diff
@@ -42,6 +43,10 @@ + (NSURL *)bundleURLForResource:(NSString *)resourceName
                   withExtension:(NSString *)resourceExtension
 {
     NSError *error;
+    if (launchingApp) {
```

Comment:
Yes, you are right. It will always rollback in ON_NEXT_RESTART mode. It should record the install mode to handle it in other way.